src/Microsoft.OpenApi.Hidi/OpenApiService.cs

```diff
@@ -263,11 +256,15 @@ private static async Task<Stream> GetStream(string input, ILogger logger)
                     {
                       DefaultRequestVersion = HttpVersion.Version20
                     };
-                    stream = await httpClient.GetStreamAsync(input);
+                    stream = await httpClient.GetStreamAsync(input, cancellationToken);
                 }
                 catch (HttpRequestException ex)
                 {
-                    logger.LogError($"Could not download the file at {input}, reason{ex}");
+#if DEBUG
+                    logger.LogCritical(ex, "Could not download the file at {inputPath}, reason: {exMessage}", input, ex.Message);
```

Comment:
```suggestion
                    throw new InvalidOperationException($"Could not download the file at {inputPath}, ex);
```
And clean the rest, replicate for a local file. Since this is a private method, and since we now have global catches that handle properly logging any exception, it's better to let the exception bubble up. (less code duplication, debugger will be able to stop here, etc...)